examples/llm/common/content_extractor_module.py

```diff
@@ -30,62 +36,122 @@
 logger = logging.getLogger(__name__)
 
 
-def read_generic_file(file):
-    """
-    Reads the content of a generic text file.
+@dataclass
+class FileMeta:
+    file_path: str
+    file_name: str
+    file_type: str
 
-    Parameters
-    ----------
-    file : fsspec.core.OpenFile
-        An instance of OpenFile representing a text file.
 
-    Returns
-    -------
-    str or None
-        The content of the file as a string, or None if an error occurs.
+class CsvTextConverter(BaseConverter):
+    """
+    Converts a CSV file column content to text documents.
     """
-    try:
-        with file.fs.open(file.path, 'r') as f:
-            return f.read()
-    except Exception as e:
-        logger.error(f"Error reading file {file.path}: {e}")
-        return None
-
 
-def read_pdf_file(file):
+    outgoing_edges = 1
+
+    def convert(self,
+                file_path: Path | list[Path] | str | list[str] | list[Path | str],
+                meta: typing.Optional[dict[str, typing.Any]],
+                remove_numeric_tables: typing.Optional[bool] = None,
+                valid_languages: typing.Optional[list[str]] = None,
+                encoding: typing.Optional[str] = "UTF-8",
+                id_hash_keys: typing.Optional[list[str]] = None) -> list[Document]:
+        """
+        Load a CSV file and convert it to Documents.
+
+        Parameters
+        ----------
+        file_path:
+            Path to the CSV file you want to convert.
+        meta:
+            A dictionary of metadata key-value pairs that you want to append to the returned document. It's optional.
+        encoding:
+            Specifies the file encoding. It's optional. The default value is `UTF-8`.
+        id_hash_keys:
+            Generates the document ID from a custom list of strings that refer to the document's attributes.
+        remove_numeric_tables: bool
+            Removes numeric tables from the csv.
+        valid_languages:
+            Valid languages
+
+        Returns
+        -------
+        list[haystack.Document]
+            List of documents, 1 document per line in the CSV.
+        """
+        if not isinstance(file_path, list):
+            file_path = [file_path]
+
+        docs: list[Document] = []
+        if meta is None:
+            text_column_name = "page_content"
```

Comment:
```suggestion
            text_column_name = "content"
```

For a CSV file, 'page_content' doesn't make sense.